### DISCUSSION TITLE
More customizable routing

### DISCUSSION THREAD

**Discussion by 3UR** [laravel/framework#50000] [Ideas]
It would be great if it was possible to change if routes are capital sensitive instead of just forcing it to be sensitive, some codebases are 13+ years old and have to communicate with APIs etc and the current routing really doesnt make it easy on huge old code bases with inconsistent casing

**Comment by henzeb** [comment]
Could you give an example on how you had this in mind?

**Reply by 3UR** [reply]
I don't have an example but I can for sure give it a try tomorrow, What I think would be good is if its defined in env maybe "SENSITIVE_ROUTING" or something like so and then if enabled routing will be capital sensitive, by default this should be enabled as projects may rely on routing being sensitive while others don't

**Reply by henzeb** [reply]
/my/Office/Api/WhErE/I/wantTo/go is no different to  /my/office/api/where/i/wantto/go... even for older api's.

or are those systems designed in a way that 'user' gets you regular user object, but 'USER' makes sure it returns an admin in a user format response?

**Reply by 3UR** [reply]
> /my/Office/Api/WhErE/I/wantTo/go is no different to /my/office/api/where/i/wantto/go... even for older api's.

Exactly, This is why I am suggesting some kind of ENV/Setting to enable or disable case sensitive routing

https://www.laravel.app/this-is-a-route -> 200 (defined in web/routes.php with the same casing)
https://www.laravel.app/tHis-iS-a-RoUtE -> 404 (defined in web/routes.php with the same exact string but different casing)

I don't see why this is limited and we have case sensitive only a option for this would be great maybe not everyone would use it but some people would use it I'm sure

Not sure if I understood what you said either but this is the best I can explain

**Reply by henzeb** [reply]
Now I know what you mean. Laravel routing is case sensitive and some api's capitalize their calls, so instead of adding routes with capitalization based on those requests, a case insensitive option is wanted to prevent 'forgotten' routes.

**Reply by ghost** [reply]
Until then maybe this could help https://serverfault.com/questions/177647/convert-and-redirect-url-in-uppercase-to-lowercase-using-htaccess

**Reply by 3UR** [reply]
>Now I know what you mean. Laravel routing is case sensitive and some api's capitalize their calls, so instead of adding routes with capitalization based on those requests, a case insensitive option is wanted to prevent 'forgotten' routes.

Yep, I feel like having something like this would be a nice addition to the framework that I'm sure a lot of people would like and use and having a option for case sensitive routes enabled by default will keep the old behavior so that way no one will have issues if their routes suddenly are no longer case sensitive and people who need routing non case sensitive can just change the config that way everyone is happy :)

**Reply by 3UR** [reply]
>Until then maybe this could help https://serverfault.com/questions/177647/convert-and-redirect-url-in-uppercase-to-lowercase-using-htaccess

I'll check this out but it would be great to have it as a official option in the framework
